starter-code/Bookshelf/app/src/main/java/ly/generalassemb/drewmahrt/bookshelf/MainActivity.java

```diff
@@ -52,13 +55,49 @@ public long getItemId(int position) {
             @Override
             public View getView(int position, View convertView, ViewGroup parent) {
                 //TODO: Update the view
+                if (convertView == null) {
+                    LayoutInflater inflater = LayoutInflater.from(MainActivity.this);
+                    convertView = inflater.inflate(android.R.layout.simple_list_item_2, null);
+                }
+                TextView textView = (TextView) convertView.findViewById(android.R.id.text1);
+                textView.setText(mBookList.get(position).getTitle());
+                TextView textView2 = (TextView) convertView.findViewById(android.R.id.text2);
+                textView2.setText(mBookList.get(position).getAuthor());
+                return convertView;
             }
         };
 
         //TODO: Set the ListView's adapter
+        mListView = (ListView) findViewById(R.id.ListView);
+        mListView.setAdapter(mBookAdapter);
+
+        mListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int position, long id) {
+                TextView textView = (TextView) view.findViewById(android.R.id.text1);
+                if(textView.getCurrentTextColor() == Color.RED){
```

Comment:
Just to get you thinking - what if you added a boolean variable to the `Book` class to keep track of whether it's red or not. Could you use `position` here to access the relevant `Book` object and use that to determine what should happen to the color? This way totally works too, but what if you wanted to have more than two different colors?